extensions/notification/configManager/MongoConfigManager.js

```diff
@@ -91,49 +91,49 @@ class MongoConfigManager extends BaseConfigManager {
 
     /**
      * Connects to MongoDB using the MongoClientInterface
-     * and retreives the metastore collection
+     * and retrieves the metastore collection
      * @param {Function} cb callback
      * @returns {undefined}
      */
-    _setupMongoClient(cb) {
+    async _setupMongoClient(cb) {
         const mongoUrl = constructConnectionString(this._mongoConfig);
-        MongoClient.connect(mongoUrl, {
+        let client = new MongoClient(mongoUrl, {
             replicaSet: this._mongoConfig.replicaSet,
             useNewUrlParser: true,
-        },
-        (err, client) => {
-            if (err) {
-                this._logger.error('Could not connect to MongoDB', {
-                    method: 'MongoConfigManager._setupMongoClient',
-                    error: err.message,
-                });
-                return cb(err);
-            }
+            readPreference: this._mongoConfig.readPreference,
+        });
+
+        try {
+            client = await client.connect();
```

Comment:
should not replace the client variable : use a different variable or single statement/varialbe (`const client = new MongoClient().connect()`)